File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    
    let result = 0;
    let st = [...ingredient];
    let idx = 0;
    
    for(let i of ingredient){
        st[idx++] = i;
        if(idx >= 4 && 
          st[idx - 4] == 1 && st[idx - 3] == 2 && st[idx - 2] == 3 && st[idx - 1] == 1) {
            result++;
            idx -= 4;
        }
    }
    
    return result;
}

File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.*;

class Solution {
    public int solution(int[] ingredient) {
        
        int result = 0;
        int[] st = new int[ingredient.length];
        int idx = 0;
        for(int i : ingredient)
        {
            st[idx++] = i;
            if(idx >= 4 && 
               st[idx - 1] == 1 && st[idx - 2] == 3 && st[idx - 3] == 2 && st[idx - 4] == 1) {
                result++;
                idx -= 4;
            }
        }
        
        return result;
    }
}